security/self-assessment.md

```diff
@@ -0,0 +1,228 @@
+- **Replicated PV Mayastor IO Engine metrics exporter:** This exposes volume controller stats data in prometheus-compatible format. This communicates with IO engines using intra Pod IPC.
+- **Replicated PV Mayastor Stats and Call-home plugin:** This is a plugin for reporting anonymous usage data from the Kubernetes cluster. It communicates with the Kubernetes API, and the Mayastor storage API to collect data.
+- **Clients:** This actor interacts with an OpenEBS cluster using standard Kubernetes tools and/or specialised clients for accessing storage layer functionality. This is usually a Kubernetes cluster admin or a storage admin.
+
+### Actions
+
+#### PVC-PV Based Volume Operations
+
+- **Volume Provisioning, De-provisioning, Expansion, and Snapshot Handling:**
+  - **What Happens:**  
+    - OpenEBS registers its provisioner plugin names with the Kubernetes API, enabling dynamic provisioning and de-provisioning of Persistent Volumes (PVs) based on Persistent Volume Claims (PVCs).
+    - The control-plane components (e.g., LocalPV storage control plane and Replicated PV Mayastor CSI Controller plugin) manage volume lifecycle events for various block and filesystem stacks.
+    - These operations extend to volume expansion and snapshot creation/management, ensuring data consistency and availability.
+  - **Security Considerations:**  
+    - Access to these operations is governed by Kubernetes RBAC. Only clients with appropriate permissions can trigger these actions.
+    - The control-plane components leverage Kubernetes primitives such as NodeAffinityLabels (for LocalPV) and VolumeAttachments (for Replicated PV Mayastor) to ensure volumes are pinned to a specific node and to enforce exclusive access.
+  
+- **Volume Access Control and Single-Tenancy:**
+  - **What Happens:**  
+    - The Replicated PV CSI plugins implement access control measures using CSI volume mode settings (e.g., `SINGLE_NODE_WRITER`) to ensure that a volume is attached exclusively to a single node.
+    - NVMe Reservations are used for Replicated PV Mayastor to enforce single-tenancy at the block device level.
+  - **Security Considerations:**  
+    - These mechanisms prevent unauthorized concurrent access, reducing the risk of data corruption or leakage.
+    - They ensure that volume access is strictly controlled and isolated at the hardware level, reinforcing the overall security posture of the storage service.
+
+#### Privileged Node-Level Operations
+
+- **Privileged Container Operations:**
+  - **What Happens:**  
+    - Node-level plugins, including those for LocalPV and Replicated PV Mayastor, run as privileged containers. This enables them to access system-level OS APIs and execute low-level operations such as direct I/O on block devices and hostPath mounts.
```

Comment:
LocalPV --> Local PV. Let's follow this throughout the docs.

Reply:
Hmmm, okay.